lighthouse-treemap/app/src/main.js

```diff
@@ -173,32 +225,88 @@ class TreemapViewer {
     return viewModes;
   }
 
-  render() {
-    TreemapUtil.walk(this.currentTreemapRoot, node => {
-      // @ts-ignore: webtreemap will store `dom` on the data to speed up operations.
-      // However, when we change the underlying data representation, we need to delete
-      // all the cached DOM elements. Otherwise, the rendering will be incorrect when,
-      // for example, switching between "All JavaScript" and a specific bundle.
-      delete node.dom;
-
-      // @ts-ignore: webtreemap uses `size` to partition the treemap.
-      node.size = node[this.currentViewMode.partitionBy || 'resourceBytes'] || 0;
-    });
-    webtreemap.sort(this.currentTreemapRoot);
+  /**
+   * @param {LH.Treemap.Selector} selector
+   */
+  setSelector(selector) {
+    this.selector = selector;
+
+    if (selector.type === 'group') {
+      this.currentTreemapRoot =
+        this.wrapNodesInNewRootNode(this.depthOneNodesByGroup[selector.value]);
+    } else if (selector.type === 'depthOneNode') {
+      let node;
+      outer: for (const depthOneNodes of Object.values(this.depthOneNodesByGroup)) {
+        for (const depthOneNode of depthOneNodes) {
+          if (depthOneNode.name === selector.value) {
```

Comment:
if there any any duplicates, then this can match incorrectly. 
with the current data, it seems unlikely, but once we go intra-bundle, i suspect we'll run into this.

Reply:
this is just depth one nodes, so for a duplicate to be possible that would mean there are multiple instances of network requests with the same URL.